File: hello_world.c
#include <stdio.h>
#include <stdlib.h>
int main() {
   printf("C: Boom-de-ah-da\n");
   fflush(stdout);  // The next program was printing first?!
   system("gcc hello_world.cpp -lstdc++ -o hello_world_cpp");
   system("hello_world_cpp");
   printf("C: Boom-de-ah-da!\n");
   return 0;
}

File: hello_world.rs
use std::process::{Command, Stdio};

fn main() {
    println!("Rust: And all its bytes and bits!");
    Command::new("gcc")
        .args(&["hello_world.c", "-o", "hello_world_c"])
        .output()
        .expect("failed to execute process");
    Command::new("hello_world_c")
        .stdout(Stdio::inherit())
        .output()
        .expect("failed to execute process");
    println!("Rust: Boom-de-ah-da!");
}
